### PR TITLE
fix(accordion): add headline to wrap panel button

### PR DESCRIPTION
**Pull request by rachelbt**

### CODE REVIEW

**Review by yinonov — COMMENTED** [review]
Are formatting changes caused by our lint script?

**Comment by rachelbt** [comment]
> Are formatting changes caused by our lint script?

I don't know....
the test file also had "only" in it

File: components/expansion-panel/readme.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 | `noRipple`         | `noRipple`         | `boolean`         | false     |                                                                                            |
 | `open`             | `open`             | `boolean`         | false     |                                                                                            |
 | `ripple`           |                    | `Promise<Ripple>` |           |                                                                                            |
+| `header-level`     | `3`                | `string` | "3"       | The expension panel headline - default is `H3` but need to set according to the page hierarchy      |
```
Comment (yinonov) [inline components/expansion-panel/readme.md:16]:
```suggestion
| `heading-level`     | `3`                | `string` | "3"       | The expension panel headline - default is `H3` but need to set according to the page hierarchy      |
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/expansion-panel/stories/arg-types.js [hunk 2]
```diff
@@ -4,6 +4,12 @@ export const argTypes = {
 			type: 'text'
 		}
 	},
+	'header-Level': {
```
Comment (yinonov) [inline components/expansion-panel/stories/arg-types.js:7]:
```suggestion
	'heading-Level': {
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/expansion-panel/src/vwc-expansion-panel.ts [hunk 3]
```diff
@@ -23,85 +23,103 @@ declare global {
...
 	@queryAsync('mwc-ripple') ripple!: Promise<Ripple>;
-
 	protected rippleHandlers = new RippleHandlers(() => {
 		return this.ripple;
 	});
-
-	protected toggleOpen(): void {
-		this.open = !this.open;
-	}
+	@property({type: String, reflect: true, attribute: 'header-level'})
```
Comment (rachelbt) [inline components/expansion-panel/src/vwc-expansion-panel.ts:66]:
```suggestion
@property({type: String, reflect: true, attribute: 'heading-Level'})
```

File: components/expansion-panel/readme.md [hunk 4]
```diff
@@ -13,6 +13,7 @@
 | `noRipple`         | `noRipple`         | `boolean`         | false     |                                                                                            |
 | `open`             | `open`             | `boolean`         | false     |                                                                                            |
 | `ripple`           |                    | `Promise<Ripple>` |           |                                                                                            |
+| `heading-level`     | `3`                | `string` | "3"       | The expension panel headline - default is `H3` but need to set according to the page hierarchy      |
```
Comment (yinonov) [inline components/expansion-panel/readme.md:16]: Attribute is still wrong

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/expansion-panel/src/vwc-expansion-panel.ts [hunk 5]
```diff
@@ -23,71 +23,79 @@ declare global {
 const iconSets = ['chevron', 'binary'];
 export type IndicatorIconSets = typeof iconSets;
 
+const VALID_HEADER_VALUES = [2, 3, 4, 5, 6];
```
Comment (yinonov) [inline components/expansion-panel/src/vwc-expansion-panel.ts:26]: Why don't we allow h1?

**Review by yinonov — CHANGES_REQUESTED** [review]
Notice that there are many formatting changes which blur the actual changes. Would make it easier to review without it

File: components/expansion-panel/readme.md (hunk 4 above)
Comment (rachelbt) [reply components/expansion-panel/readme.md]: the attribute is OK the description isn't. Fixed it.

File: components/expansion-panel/src/vwc-expansion-panel.ts (hunk 5 above)
Comment (yinonov) [reply components/expansion-panel/src/vwc-expansion-panel.ts:26]: fair enough

File: components/expansion-panel/readme.md [hunk 6]
```diff
@@ -13,7 +13,7 @@
 | `noRipple`         | `noRipple`         | `boolean`         | false     |                                                                                                                                                  |
 | `open`             | `open`             | `boolean`         | false     |                                                                                                                                                  |
 | `ripple`           |                    | `Promise<Ripple>` |           |                                                                                                                                                  |
-| `heading-level`     | `3`                | `string` | "3"       | The expension panel `heading-level` default is `3` (meaning the button is wrapped with `H3` tag) but need to set according to the page hierarchy |
+| `headerLevel`     | `heading-level`                | `string` | "3"       | The expension panel `heading-level` default is `3` (meaning the button is wrapped with `H3` tag) but need to set according to the page hierarchy |
```
Comment (yinonov) [inline components/expansion-panel/readme.md:16]:
```suggestion
| `headingLevel`     | `heading-level`                | `string` | "3"       | The expansion panel `heading-level` default is `3` (meaning the button is wrapped with `H3` tag) but need to set according to the page hierarchy |
```

File: components/expansion-panel/src/vwc-expansion-panel.ts [hunk 7]
```diff
@@ -23,71 +23,79 @@ declare global {
...
-
 	protected rippleHandlers = new RippleHandlers(() => {
 		return this.ripple;
 	});
-
-	protected toggleOpen(): void {
-		this.open = !this.open;
-	}
+	@property({type: String, reflect: true, attribute: 'heading-level'})
+	private headerLevel = '3';
```
Comment (yinonov) [inline components/expansion-panel/src/vwc-expansion-panel.ts:68]:
```suggestion
	private headingLevel = '3';
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/expansion-panel/readme.md [hunk 8]
```diff
@@ -2,17 +2,18 @@
...
+| `leadingToggle`    | `leadingToggle`    | `boolean`         | false     |                                                                                                                                                  |
+| `noRipple`         | `noRipple`         | `boolean`         | false     |                                                                                                                                                  |
+| `open`             | `open`             | `boolean`         | false     |                                                                                                                                                  |
+| `ripple`           |                    | `Promise<Ripple>` |           |                                                                                                                                                  |
+| `headingLevel`    | `heading-level`                | `string` | "3"       | The expension panel `heading-level` default is `3` (meaning the button is wrapped with `H3` tag) but need to set according to the page hierarchy |
```
Comment (yinonov) [inline components/expansion-panel/readme.md:16]:
```suggestion
| `headingLevel`    | `heading-level`                | `string` | "3"       | The expension panel `heading-level` defaults to `3` (meaning the button is wrapped with a `H3` tag) but need to be in accordance to the page's hierarchy |
```

**Review by yinonov — APPROVED** [review]

**Pull request merged**